wire-ios/Wire-iOS/Sources/URLActionRouter.swift

```diff
@@ -133,6 +133,46 @@ class URLActionRouter: URLActionRouterProtocol {
 // MARK: - PresentationDelegate
 extension URLActionRouter: PresentationDelegate {
 
+    func showPasswordPrompt(for conversationName: String, completion: @escaping (String?) -> Void) {
+        typealias ConversationAlert = L10n.Localizable.Join.Group.Conversation.Alert
+
+        let alertController = UIAlertController(
+            title: ConversationAlert.title(conversationName),
+            message: ConversationAlert.message,
+            preferredStyle: .alert
+        )
+
+        alertController.addTextField { textField in
+            textField.placeholder = ConversationAlert.Textfield.placeholder
+            textField.isSecureTextEntry = true
+        }
+
+        let joinAction = UIAlertAction(title: ConversationAlert.JoinAction.title, style: .default) { _ in
+            let password = alertController.textFields?.first?.text
+            completion(password)
+        }
+
+        let helpLinkURL = URL(string: "https://support.wire.com/hc/en-us/articles/360000574069-Create-guest-rooms")!
+        let learnMoreAction = UIAlertAction(title: ConversationAlert.LearnMoreAction.title, style: .default) { _ in
+            UIApplication.shared.open(helpLinkURL, options: [:], completionHandler: nil)
+        }
+
+        let cancelAction = UIAlertAction(title: L10n.Localizable.General.cancel, style: .cancel) { _ in
+            completion(nil)
+        }
+
+        alertController.addAction(joinAction)
+        alertController.addAction(learnMoreAction)
+        alertController.addAction(cancelAction)
+
+        // Use the rootViewController to present the alert
+        if delegate?.urlActionRouterCanDisplayAlerts() ?? true {
```

Comment:
Whew that's interesting to evaluate! Is it the same as this? Hm 🤔
```suggestion
        if delegate == nil || delegate?.urlActionRouterCanDisplayAlerts() == true {
```